pkg/certificate/controller.go

```diff
@@ -43,19 +43,43 @@ func (m *Manager) add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return object.GetName() == m.webhookName
 	}
 
+	isGeneratedSecret := func(object client.Object) bool {
+		if object.GetName() == m.caSecretKey().Name {
+			return true
+		}
+
+		webhookConf, err := m.readyWebhookConfiguration()
+		if err != nil {
+			m.log.Info(fmt.Sprintf("failed checking if it's a generated secret: failed getting webhook configuration: %v", err))
+			return false
+		}
+
+		services, err := m.getServicesFromConfiguration(webhookConf)
+		if err != nil {
+			m.log.Info(fmt.Sprintf("failed checking if it's a generated secret: failed getting webhook configuration services: %v", err))
+			return false
+		}
+
+		for service, _ := range services {
+			if object.GetName() == service.Name {
+				return true
+			}
+		}
```

Comment:
can we move [this](https://github.com/qinqon/kube-admission-webhook/pull/60/files#diff-477009fdb255fbbfc90de2def9ee37495b402bf4e99058b8fdf409341c11f066R51-R67) to a specific function `isServiceSecretRelated` (or any better name you can find)

This way `isGeneratedSecret` will be more readable to a stranger:
```
return isCASecretRelated(object) || isServiceSecretRelated(object)
```